src/particle_restart_write.F90

```diff
@@ -40,13 +40,22 @@ subroutine write_particle_restart(p)
     src => source_bank(current_work)
 
     ! Write data to file
-    call write_dataset(file_id, 'filetype', FILETYPE_PARTICLE_RESTART)
+    call write_dataset(file_id, 'filetype', 'particle restart')
     call write_dataset(file_id, 'revision', REVISION_PARTICLE_RESTART)
     call write_dataset(file_id, 'current_batch', current_batch)
     call write_dataset(file_id, 'gen_per_batch', gen_per_batch)
     call write_dataset(file_id, 'current_gen', current_gen)
     call write_dataset(file_id, 'n_particles', n_particles)
-    call write_dataset(file_id, 'run_mode', run_mode)
+    select case(run_mode)
+    case (MODE_FIXEDSOURCE)
+      call write_dataset(file_id, 'run_mode', 'fixed source')
+    case (MODE_EIGENVALUE)
+      call write_dataset(file_id, 'run_mode', 'k-eigenvalue')
+    case (MODE_PLOTTING)
+      call write_dataset(file_id, 'run_mode', 'plot')
```

Comment:
Well, plotting does use a fake particle to find cells, but I think it's safe to say no, plotting mode should never produce a particle restart file.